src/actions/reportArticle.js

```diff
@@ -0,0 +1,57 @@
+import axios from "axios";
+import * as types from "../actionTypes/reportArticle";
+
+const reportArticleSucess = reportResponse => ({
+  type: types.REPORT_ARTICLE_SUCCESS,
+  payload: reportResponse
+});
+
+const reportArticleError = error => ({
+  type: types.REPORT_ARTICLE_ERROR,
+  payload: error
+});
+
+const reportArticleLoading = loading => ({
+  type: types.REPORT_ARTICLE_LOADING,
+  payload: loading
+});
+
+const reportArticle = (
+  reasonForReport,
+  reportBody,
+  articleslug
+) => dispatch => {
+  const { token } = localStorage;
+
+  dispatch(reportArticleLoading(true));
+  const { REACT_APP_API } = process.env;
+  return axios
+    .post(
+      `${REACT_APP_API}/api/articles/${articleslug}/report`,
+      { reasonForReport, reportBody },
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json"
+        }
+      }
+    )
+    .then(response => {
+      if (response.data.status === "success") {
+        dispatch(reportArticleLoading(false));
+        return dispatch(reportArticleSucess(response.data));
+      }
+      dispatch(reportArticleLoading(false));
```

Comment:
line 41 and line 44 can be replaced with a single dispatch before the if statement on line 40.